tests/providers/apache/hive/hooks/test_hive.py

```diff
@@ -687,6 +687,13 @@ def test_get_pandas_df(self):
         hook.mock_cursor.execute.assert_any_call('set airflow.ctx.dag_owner=airflow')
         hook.mock_cursor.execute.assert_any_call('set airflow.ctx.dag_email=test@airflow.com')
 
+        hook = MockHiveServer2Hook(empty_table_flag = True)
```

Comment:
I’d probably do something like `MockHiveServer2Hook(connection_cursor=EmptyMockTableCursor())` instead. `EmptyMockTableCursor` would inherit from a new `BaseMockHiveServer2Hook` class that contains common logic refactored out of `MockHiveServer2Hook` (which would then also inherit from `BaseMockHiveServer2Hook`).

Reply:
Let's see how these latest changes work. I removed the `empty_table_flag ` flag & split up the `MockConnectionCursor` class.  It does look better.